HW6_project/Program.cs

```diff
@@ -2,8 +2,311 @@
+        result[result.Length - 1] = chosenElement;
+        for (int i = 0; i < array.Length; i++)
+        {
+            result[i] = array[i];
+        }
+
+        return result;
+    }
+    //ÚLOHA 8
+    public static int[] AddElementOnIndex(int[] array)
+    {
+        Console.WriteLine("ADD ELEMENT ON INDEX");
+        Console.WriteLine("Choose on what index you want to add an element");
+        string ReadIndex = Console.ReadLine();
+        Int32.TryParse(ReadIndex, out int chosenIndex);
+        if (chosenIndex > array.Length || chosenIndex < 0)
+        {
+            Console.WriteLine("Index out of Bounds");
+            return array;
+        }
+
+        Console.WriteLine("Choose an element you want to add");
+        string ReadElement = Console.ReadLine();
+        Int32.TryParse(ReadElement, out int chosenElement);
+
+        int[] result = new int[array.Length + 1];
+        result[chosenIndex] = chosenElement;
+        int indexSkip = 0;
+
+        for (int i = 0; i < array.Length; i++)
+        {
+            if (i == chosenIndex)
+            {
+                indexSkip++;
+            }
+            result[i + indexSkip] = array[i];
+        }
+
+        return result;
+    }
+    public static int[] AddElementOnIndex(int[] array, int chosenIndex)
+    {
+
+        Console.WriteLine("ADD ELEMENT ON INDEX");
+        if (chosenIndex > array.Length || chosenIndex < 0)
+        {
+            Console.WriteLine("Index out of Bounds");
+            return array;
+        }
+
+        Console.WriteLine("Choose an element you want to add");
+        string ReadElement = Console.ReadLine();
+        Int32.TryParse(ReadElement, out int chosenElement);
+
+        int[] result = new int[array.Length + 1];
+        result[chosenIndex] = chosenElement;
+        int indexSkip = 0;
+
+        for (int i = 0; i < array.Length; i++)
+        {
+            if (i == chosenIndex)
+            {
+                indexSkip++;
+            }
+            result[i + indexSkip] = array[i];
+        }
+
+        return result;
+    }
+    public static int[] AddElementOnIndex(int[] array, int chosenIndex, int chosenElement)
+    {
+
+        Console.WriteLine("ADD ELEMENT ON INDEX");
+        if (chosenIndex > array.Length || chosenIndex < 0)
+        {
+            Console.WriteLine("Index out of Bounds");
+            return array;
+        }
+
+        int[] result = new int[array.Length + 1];
+        result[chosenIndex] = chosenElement;
+        int indexSkip = 0;
+
+        for (int i = 0; i < array.Length; i++)
+        {
+            if (i == chosenIndex)
+            {
+                indexSkip++;
+            }
+            result[i + indexSkip] = array[i];
+        }
+
+        return result;
+    }
+
+
+    //Pomocné funkce
+    public static void WriteArrayFirstToLast(Array array)
+    {
+        for (int i = 0; i < array.Length; i++)
+        {
+            Console.Write(array.GetValue(i) + ", "); //Tohle jsem si musel najít :D
```

Comment:
Ono sis pomihal pouzivani `int[]` a `Array`, kdybys typ argumentu v definici parametru funkce vymenil (na radku 294 dal int[] array), tak bys tady mohl pouzit normalne toto:
```suggestion
            Console.Write(array[i] + ", ");
```

coz uz je bezny pouziti, co jsme si ukazovali :D